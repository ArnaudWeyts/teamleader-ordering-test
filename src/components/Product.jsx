import React from 'react';

import { formatToPrice } from '../helpers';

const Product = ({ product }) =>
  !product ? (
    <div>Loading...</div>
  ) : (
    <div>
      <p>Description: {product.description}</p>
      <p>Category: {product.category}</p>
      <p>Price: {formatToPrice(product.price)}</p>
    </div>
  );

export default Product;
