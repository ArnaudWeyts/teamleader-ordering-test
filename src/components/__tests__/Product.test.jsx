import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Product from '../Product';

it('renders without crashing', () => {
  shallow(<Product />);
});

it('renders a product as loading', () => {
  const component = renderer.create(<Product />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a product', () => {
  const product = {
    id: 'A101',
    description: 'Screwdriver',
    category: '1',
    price: 9.75
  };

  const component = renderer.create(<Product product={product} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
