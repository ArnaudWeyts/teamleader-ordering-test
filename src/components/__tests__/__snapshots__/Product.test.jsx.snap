// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a product 1`] = `
<div>
  <p>
    Description: 
    Screwdriver
  </p>
  <p>
    Category: 
    1
  </p>
  <p>
    Price: 
    â‚¬9.75
  </p>
</div>
`;

exports[`renders a product as loading 1`] = `
<div>
  Loading...
</div>
`;
